src/Cli/Utils.cs

```diff
@@ -904,27 +904,37 @@ public static bool WriteJsonContentToFile(string file, string jsonContent)
         /// This method will check if DAB_ENVIRONMENT value is set.
         /// If yes, it will try to merge dab-config.json with dab-config.{DAB_ENVIRONMENT}.json
         /// </summary>
-        public static string MergeConfigsIfAvailable()
+        /// <returns>Returns the name of the merged Config if successful.</returns>
+        public static bool TryMergeConfigsIfAvailable(out string mergedFile)
         {
             string? environmentValue = Environment.GetEnvironmentVariable(RUNTIME_ENVIRONMENT_VAR_NAME);
+            mergedFile = string.Empty;
             if (!string.IsNullOrEmpty(environmentValue))
             {
-                string FileToBeOverridden = DefaultName;
-                string FileToBeOverriddenWith = $"{CONFIGFILE_NAME}.{environmentValue}{CONFIG_EXTENSION}";
-                string mergedFile = $"{CONFIGFILE_NAME}.{environmentValue}.merged{CONFIG_EXTENSION}";
+                string baseConfigFile = DefaultName;
+                string overrideConfigFile = RuntimeConfigPath.GetFileName(environmentValue, considerOverrides: false);
```

Comment:
make sure to document this. Merging config files explicitly is incompatible with override config files. This would be a breaking change in the specific case when these files exist in the engine's exe directory, wouldn't it?:
1. dab-config.json
2. dab-config.mssql.json
3. dab-config.mssql.overrides.json (no longer used, as 1 and 2 are merged.)

Reply:
it's not a breaking change. since we are not removing any existing feature, dab-override files can still be used. for the scenario where we have all the files and dab environment is not set up, but hosting environment is set to `mssql`. then in that case no merging will happen, and being highest in precedence dab-config.mssql.overrides.json would be used.

Reply:
But I'll update the docs once this change gets approved as now the docs are moved in a different repo.

Reply:
added more comments in the code.